src/main/ruby/core/io.rb

```diff
@@ -1184,13 +1185,15 @@ def initialize(fd, mode=nil, options=undefined)
 
     @eof = false
 
-    mode, binary, external, internal, @autoclose = IO.normalize_options(mode, options)
+    mode, binary, external, internal, autoclose_tmp = IO.normalize_options(mode, options)
 
     IO.setup self, Rubinius::Type.coerce_to(fd, Integer, :to_int), mode
 
     binmode if binary
     set_encoding external, internal
 
+    @autoclose = autoclose_tmp
```

Comment:
Why is the variable assigned later?

Reply:
@aardvark179 Or is for readability?

Reply:
`IO.setup` sets `@autoclose` to a default value because it sets everything, so if we want to set it to anything other than true this has to be done after the `IO.setup` call.

I think in the long term `IO.setup` should take autoclose as an additional option but I wanted to go through the IO code and make sure I understood all the code paths by which IO objects are created before doing that.

Reply:
I see. Would definitely be worth a comment :)
Could you clean it up soon?
Now that we have our own fork of the socket stuff it should be easy to modify all callers of `IO.setup`.